### DISCUSSION TITLE
Feature Request: Support AWS ECR authentication scheme in GitHub Actions container

### DISCUSSION THREAD

**Discussion by el-pendeloco** [community/community#4770] [Actions]
Currently, GitHub Actions supports setting static credentials for jobs containers: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainercredentials.

AWS ECR credentials, however, are temporary: https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html.

In order to download private ECR repository images to run our actions, it seems we have some very standard fiddling to do to generate those credentials and then pass them on to the job (run some commands to get temporary credentials, set them as output and reuse them in the later step, is one way to do it).

CircleCI, for example, automates this with a simple "aws_auth" configuration block for the container: https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html.

I think it would be nice to have this automated in Github Actions. I think it would encourage adoption for those of us who make use of private ECR repositories in our CI/CD pipelines.

**Comment by LeKristapino** [comment]
This would be a very useful feature - our organization uses ECR a lot (due to lower costs than in Dockerhub and credits) and it is very annoying that there is no straight-up way to use Private ECR repositories as Github actions containers. This is the sole reason why we are currently thinking of looking for an alternative CI/CD platform.

**Comment by nanonyme** [comment]
This is also  quite inconvenient for GitHub Enterprise users. Even when you're hosting your self-hosted runners in AWS, there is no good solution for using ECR with GitHub Actions using OIDC authentication.
